state.go

```diff
@@ -257,6 +258,37 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		}
 	}
 
+	// Also make an attempt to contact the node directly over TCP. This
+	// helps prevent confused clients who get isolated from UDP traffic
+	// but can still speak TCP (which also means they can possibly report
+	// misinformation to other nodes via anti-entropy), avoiding flapping in
+	// the cluster.
+	fallbackCh := make(chan bool)
+	if node.PMax >= 3 {
+		destAddr := &net.TCPAddr{IP: node.Addr, Port: int(node.Port)}
```

Comment:
It's a little dirty to have this net stuff up here, but I want to silently move on if the connect fails. If this is down inside sendPingAndWaitForAck then it complicates the return values (you need to be able to say "there were no errors but we wen't able to get a ping and an ack".